app/misc/graphing.ts

```diff
@@ -385,11 +415,43 @@ function makeBasicNode(c, column: number) {
             });
 
             bsEdges.add({
-                from: id + numOfCommits * (i + 1),
+                from: bsnodeId,
                 to: id
             });
 
-            branchIds[id] = id + numOfCommits * (i + 1);
+            branchIds[id] = bsnodeId;
+        }
+    }
+
+    // Initializing viewable tags in highest zoom graph level
+    if (c.toString() in tags) {
+        for (let i = 0; i < tags[c.toString()].length; i++) {
+            let tagName = tags[c.toString()][i];
+            let tp = tagName.name().split("/");
+            let shortTagName = tp[tp.length - 1];
+            console.log(shortTagName + " tag: " + tagName.isHead().toString());
+            if (tagName.isHead()) {
+                shortTagName = "*" + shortTagName;
+            }
+            let bsnodeId = generateUniqueNumber();
+            bsNodes.add({
+                id: bsnodeId,
+                shape: "ellipse",
+                // color: "teal",
+                title: tagName, // hover text
+                label: shortTagName, // shown under/in shape
+                physics: false,
+                fixed: false,
+                x: (column - 0.6 * (i + 1)) * tagSpacingX,
```

Comment:
A comment explaining this math would be nice.

Reply:
Again, I'm not entirely sure what the math is doing other than giving the node a specific starting spot. Would that be an appropriate comment?

Reply:
Yep! Just saying what it does conceptually.